angular/core/components/change_volume_form/change_volume_form.coffee

```diff
@@ -0,0 +1,42 @@
+angular.module('loomioApp').factory 'ChangeVolumeForm', ->
+  templateUrl: 'generated/components/change_volume_form/change_volume_form.html'
+  controller: ($scope, model, setDefault, FormService, CurrentUser, FlashService) ->
+    $scope.model = model.clone()
+    $scope.volumeLevels = ["loud", "normal", "quiet"]
+    $scope.buh = {}
+    $scope.buh.volume = model.volume()
+    $scope.setDefault = setDefault if setDefault
+
+    $scope.translateKey =
+      if setDefault
+        "change_volume_form.set_membership_default"
+      else
+        "change_volume_form.#{$scope.model.constructor.singular}"
+
+    $scope.title =
+      if setDefault
+        ''
+      else
+        model.title or model.groupName()
+
+
+    $scope.submit = ->
+      $scope.isDisabled = true
```

Comment:
yea please!

On Tue, 22 Mar 2016 at 09:41 James Kiesel <notifications@github.com> wrote:

> In angular/core/components/change_volume_form/change_volume_form.coffee
> <https://github.com/loomio/loomio/pull/3130#discussion_r56892950>:
>
> > +
> > +    $scope.translateKey =
> > +      if setDefault
> > +        "change_volume_form.set_membership_default"
> > +      else
> > +        "change_volume_form.#{$scope.model.constructor.singular}"
> > +
> > +    $scope.title =
> > +      if setDefault
> > +        ''
> > +      else
> > +        model.title or model.groupName()
> > +
> > +
> > +    $scope.submit = ->
> > +      $scope.isDisabled = true
>
> Sweet; shall I put an entry into docs/en/somewhere titled 'How to use
> FormService'?
>
> —
> You are receiving this because you commented.
>
>
> Reply to this email directly or view it on GitHub
> <https://github.com/loomio/loomio/pull/3130/files/06bc4696156c7b19ed6dd2d6a8dc9e26f8ab3966#r56892950>
>